src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace RebelCode\Bookings\WordPress\Module;
+
+use Dhii\Data\Container\ContainerFactoryInterface;
+use Psr\Container\ContainerInterface;
+use Psr\EventManager\EventManagerInterface;
+use RebelCode\Modular\Module\AbstractBaseModule;
+
+class WpBookingsUiModule extends AbstractBaseModule
+{
+    private $bookingsPageId;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param ContainerFactoryInterface $containerFactory The factory for creating container instances.
+     * @throws \Dhii\Exception\InternalException
+     */
+    public function __construct(ContainerFactoryInterface $containerFactory)
+    {
+        $this->_initModule(
+            $containerFactory,
+            'wp_bookings_ui',
+            ['wp_event_manager'],
+            $this->_loadPhpConfigFile(WP_BOOKINGS_UI_MODULE_DIR . '/config.php')
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setup()
+    {
+        return $this->_createContainer([
+            'template_manager' => function (ContainerInterface $c) {
+                $templateManager = new TemplateManager($c->get('wp_event_manager'));
+                $templateManager->registerTemplates($this->_getConfig()['templates']);
+                return $templateManager;
+            }
+        ]);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function run(ContainerInterface $c = null)
+    {
+        /* @var EventManagerInterface $eventManager  */
+        $eventManager = $c->get('wp_event_manager');
+
+        /** @var TemplateManager $templateManager */
+        $templateManager = $c->get('template_manager');
+
+        $eventManager->attach('admin_init', function () use ($eventManager, $templateManager) {
+            $this->_adminInit($eventManager, $templateManager);
+        });
+
+        $eventManager->attach('admin_menu', function () use ($templateManager) {
+            $this->_adminMenu($templateManager);
+        });
+    }
+
+    /**
+     * Register hook on admin init which will register everything else.
+     *
+     * @param EventManagerInterface $eventManager
+     * @param TemplateManager $templateManager
+     */
+    protected function _adminInit($eventManager, $templateManager)
+    {
+        /*
+         * Add metabox with availabilities configuration to
+         * service's edit page.
+         */
+        add_meta_box(
+            $this->_getConfig()['metabox']['id'],
+            $this->_getConfig()['metabox']['title'],
+            function () use ($templateManager) {
+                return $this->_renderMetabox($templateManager);
+            },
+            $this->_getConfig()['metabox']['post_type']

```

Comment:
Kindly optimize this code. `$this->_getConfig()` is being called 3 times and each time it is being accessed at `'metabox'`. Consider retrieving `$this->_getConfig()['metabox']` only once.